### PR TITLE
Add Github Actions Support

### PR DESCRIPTION
**Pull request by shibumi**
**Fixes issue #**:
https://github.com/in-toto/in-toto-golang/issues/62

**Description of pull request**:
This PR intents to remove travis and appveyor as CI and use Github Actions for everything, what we need.

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Comment by shibumi** [comment]
This is Work in Progress.. I am still playing around with Github Actions and Go.

**Comment by shibumi** [comment]
Small update: MacOS and Linux is working, even coverage and everything works. What doesn't work is Windows and I don't know why yet.

```yml
on: [push, pull_request]
name: Test
jobs:
  test:
    strategy:
      matrix:
        go-version: [1.14.x]
        os: [ubuntu-latest, macos-latest, windows-latest]
    runs-on: ${{ matrix.os }}
    steps:
    - name: Install Go
      uses: actions/setup-go@v2
      with:
        go-version: ${{ matrix.go-version }}
    - name: Checkout code
      uses: actions/checkout@v2
    - name: Format Unix
      if: runner.os == 'Linux' || runner.os == 'MacOS'
      run: test -z $(go fmt ./...)
    - name: Format Windows
      if: runner.os == 'Windows'
      run: if (gofmt -l ./...) { Write-Error "Use gofmt" }
    - name: Test
      run: go test -race -covermode atomic -coverprofile=profile.cov ./...
    - name: Send coverage
      env:
        COVERALLS_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      run: |
        GO111MODULE=off go get github.com/mattn/goveralls
        $(go env GOPATH)/bin/goveralls -coverprofile=profile.cov -service=github
```

This is our current workflow file. If you want to see how this look like, click here: https://github.com/shibumi/in-toto-golang/runs/1009326668?check_suite_focus=true

**Comment by shibumi** [comment]
A few open questions:

* Right now we are calculating the coverage three times.. This is not necessary.. so this might should be in an own job.
* Testing, however, should still be run on all three OS (I had issues with paths in the past, Linux has been fine, Windows has been not)
* I don't know if the 'barrier' works with the current goveralls setup.. the runner is printing an URL and that URL works.. so we can see the coverage there.

I also thought about testing with different Go versions, but our strict dependency on Go 1.15 would make this pretty lonely right now.

**Comment by shibumi** [comment]
> 'barrier'?

What I mean with 'barrier', is the PR-blocker. For example:"Coverage decreased more than 1.0%, merging is blocked".
I have no idea if this works with 3 different coverages.

**Comment by lukpueh** [comment]
> What I mean with 'barrier', is the PR-blocker. For example:"Coverage decreased more than 1.0%, merging is blocked".
I have no idea if this works with 3 different coverages.

I see. Good question. I think it sends the aggregate value. But I'm not sure. It does seem to work for python-(in-toto) (see e.g. https://coveralls.io/builds/26585333, where we have one runaway value that affects the overall result).

**Comment by shibumi** [comment]
> I see. Good question. I think it sends the aggregate value. But I'm not sure. It does seem to work for python-(in-toto) (see e.g. https://coveralls.io/builds/26585333, where we have one runaway value that affects the overall result).

We could just merge as soon I have fixed that windows build and see if the PR-blocker will appear(?) or I test this on my own repository with different branches.

**Comment by shibumi** [comment]
@lukpueh Summary for you, because I know you are rare of time :)

* Github Actions works on Windows, Ubuntu and MacOS now
* But we only run format + send coverage on Ubuntu right now. I still have a few path problems on Windows I don't get fixed. For example on Windows if I call gofmt I get permission denied errors + sending coverage fails, too. We could enable  format + coverage for MacOS, too. What's your opinion?
* First I was going for a .gitattributes file in test/data, the latest commit (https://github.com/in-toto/in-toto-golang/pull/72/commits/04b13d5eed7111c2aa21ca0c84f37008083a4dfa) removes this, via replacing all "\r\n" in a file in the RecordArtifact step via "\n". So the .gitattributes file is now longer needed. 

If we merge this now, we can be sure for later commits, that it works on all plattforms. However, I don't know if we might have similar issues with reading links(?). Are we unmarshalling the links directly?

**Comment by shibumi** [comment]
The current Github Actions Output can be found here: https://github.com/shibumi/in-toto-golang/runs/1038690604?check_suite_focus=true

I guess, this is ready to merge.

**Comment by shibumi** [comment]
We might want to remove the appveyor CI afterwards, via the appveyor controlpanel.

File: in_toto/runlib.go [hunk 1]
```diff
@@ -27,7 +28,9 @@ using sha256 and returns a map in the following format:
 	}
 
 If reading the file fails, the first return value is nil and the second return
-value is the error.
+value is the error. For multi operating system support, we normalize the
+line separators of all files. This means, that Windows CRLF line separators
+will being transformed to Linux LF line separators.
```
Comment (lukpueh) [inline in_toto/runlib.go:33]: What do you think about ...?
```suggestion
value is the error.
NOTE: For cross-platform consistency Windows-style line separators (CRLF) are
normalized to Unix-style line separators (LF) before hashing file contents.
```

Comment (shibumi) [reply in_toto/runlib.go:33]: sounds good. I will make the necessary changes.

**Comment by shibumi** [comment]
@lukpueh I have added your comment :) should be fine now.

**Review by lukpueh — APPROVED** [review]
LGTM! Thanks, @shibumi!

**Review by lukpueh — CHANGES_REQUESTED** [review]
Oh, just realised that we will also want to remove the badges in README.md, and add new ones if they exist.

**Comment by shibumi** [comment]
> Oh, just realised that we will also want to remove the badges in README.md, and add new ones if they exist.

this is a good idea. I will add them.

File: README.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # In-toto Go implementation
-[![Build Status](https://travis-ci.com/in-toto/in-toto-golang.svg?branch=master)](https://travis-ci.com/in-toto/in-toto-golang) [![Coverage Status](https://coveralls.io/repos/github/in-toto/in-toto-golang/badge.svg)](https://coveralls.io/github/in-toto/in-toto-golang) [![Build status](https://ci.appveyor.com/api/projects/status/n45pmpso0t5b40vk?svg=true)](https://ci.appveyor.com/project/in-toto/in-toto-golang)
+![build](https://github.com/in-toto/in-toto-golang/workflows/build/badge.svg) [![Coverage Status](https://coveralls.io/repos/github/in-toto/in-toto-golang/badge.svg)](https://coveralls.io/github/in-toto/in-toto-golang)
```
Comment (lukpueh) [inline README.md:2]: Can you make the first badge a link to the GitHub build page?

Comment (shibumi) [reply README.md]: @lukpueh solved, but we can only link to the build workflow overview page. A link to the latest build is not possible (Github is working on this)

Comment (lukpueh) [reply README.md]: Thanks for the fix and headsup!

Comment (lukpueh) [reply README.md]: I suppose we have to first activate something for this to work on the upstream repo?

Comment (lukpueh) [reply README.md]: Let's merge and see. :)

**Review by lukpueh — APPROVED** [review]

**Pull request merged**